File: solutions/3/java/Solution.java
import java.util.*;

public class Solution {
    public int lengthOfLongestSubstring(String s) {
        int i = 0;
        int j = 0;
        int max = 0;
        HashSet<Character> set = new HashSet<Character>();

        while (j < s.length()) {
            if (set.add(s.charAt(j))) {
                j++;
                max = Math.max(max, set.size());
            } else {
                set.remove(s.charAt(i));
                i++;
            }
        }

        return max;
    }
}

File: solutions/3/javascript/solution.js
// Find the length of the longest substring without repeating characters

const lengthOfLongestSubstring = (s) => {
	let i = 0,
		j = 0,
		max = 0;
	let aux = new Set();

	while (j < s.length) {
		if (!aux.has(s.charAt(j))) {
			aux.add(s.charAt(j));
			j++;
			max = Math.max(max, aux.size);
		} else {
			aux.delete(s.charAt(i));
			i++;
		}
	}

	return max;
};
